### ISSUE TITLE
Implementation of CKPrettyError

### ISSUE THREAD

**Issue by malhal** [BennyKJohnson/OpenCloudKit#8]
I implemented CKPrettyError and have used it in some of the operations but some more need changed, if you agree to how it has been implemented. It matches how ObjC CloudKit works, it did cross my mind that Swift errors could be done differently. Anyway how it works i the operations create a CKPrettyError using a code, and then the class overrides description to output a nice string for the error, e.g. 

`<CKError 0x618000240c60: "Operation Cancelled" (20); "Operation <TestOperation: 0x7f876f405ea0; operationID=1644E6818C660C6E, stateFlags=executing|cancelled, qos=Utility> was cancelled before it started">`

I have tried to match that format as best I could, wasn't able to get the pointer though.

**Comment by malhal** [comment]
Todo, create CKPrettyErrors in all operations

**Comment by rhunt222** [comment]
@malhal Can you provide an example of implementing CKPrettyErrors or CKError?

**Comment by malhal** [comment]
Yes I could, but you might be interested in [Hopper](https://www.hopperapp.com) which can show the source code, pricey app but well worth it.

**Comment by rhunt222** [comment]
Thanks, I'll check it out!
